lib/list.h

```diff
@@ -13,6 +13,7 @@
 #define __METAL_LIST__H__
 
 #include <stdlib.h>
+#include <stdbool.h>
```

Comment:
nitpicking: please declare it in an alphabetical order so before `#include <stdlib.h>`